### ISSUE TITLE
How to See All the Bids Returned from Prebid.js ?

### ISSUE THREAD

**Issue by millieone** [prebid/Prebid.js#221]
I tried posting this on stackoverflow but I don't have enough reputation points to create the prebid.js tag. Can someone create that tag over there and then this type of question could go there which is where I think it should go according to the support suggestions on prebid.org

I'm trying to output all the bids from all my partners onto the page for debugging via prebid.js.

If I put this in my javascript in the header:

`    var targetingParams = pbjs.getAdserverTargeting();
    var container = document.getElementById('container');
    container.innerHTML = JSON.stringify(targetingParams);`
   
and then have this in my body:
`<div id="container"></div>`
    
All the winning bids are output on the page like this
{"div-id-0":{"hb_bidder":"winner","hb_adid":"705ed574a","hb_size":"970x250","hb_pb":"2.25"},"

However its only the winning bids. I'd like to be able to see all the bids for all sizes from even losing partners to make sure everything is setup properly.

If I change the header to this (replacing pbjs.getAdserverTargeting(); with pbjs.getBidResponses();)
` var targetingParams = pbjs.getBidResponses();
    var container = document.getElementById('container');
    container.innerHTML = JSON.stringify(targetingParams);`

Then I can see all the bids somewhere on the page but also I get a ton of other stuff output that produces a ton of 404 errors. It is hard to look at.

So what is the easiest way to see all the bids on the page? I'd even settle for being able to quickly see them in the inspector.  I tried using alwaysUseBid: true, but can't get it to work.

Thanks to all the work done so far. I have a nominal programming background and have a test page up, but would like to be able to see all the bids just so I'm confident it's all working.  Also is there a place people can post code that works for them? I think that would be helpful. For example getting sizemapping to work is something we think we have figured out. Would be nice to share.

**Comment by mkendall07** [comment]
@millieone 
You can get all the bids back by calling:
`console.log(pbjs.getBidResponses());` in the callback handler (after bids are back).

I'll repost this question on SO. 

Regarding the sizemapping -if you feel like you have some working code it would be great if you can submit a PR for review. Thanks

**Comment by mkendall07** [comment]
@millieone 
Where you able to get this to work?

**Comment by millieone** [comment]
Sorry for the delay.

We just went live and are able to get things working! We were having a lot of problems with timing out even setting  var PREBID_TIMEOUT = 5000; We figured out the problem was. In addition to the async DFP tags we were also using serving one ad on the pages using the DFP passback tag `<script type='text/javascript' src='http://www.googletagservices.com/tag/js/gpt
.js'>googletag.pubads().definePassback('/XXXXXX/yyyyy', [728, 90]).display();
</script>` to serve a variable width ad this isn't a predefined size and doesn't work well with async. Once we removed the DFP passback tag we are now getting bids with TIMEOUT=1000

The `console.log(pbjs.getBidResponses());` works well thanks for that tip. 

For sizemapping we're just using a workaround. To have different ad units based on size of the browser in prebid we just do this:
`if (jQuery(window).width() >= 960) {
        var adUnits = [{`
We then have separate the traditional sizemapping for DFP but it isn't too burdensome to do both.
